plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/SecretKeySerializer.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.exchange;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+
+import javax.crypto.SecretKey;
+
+import java.io.IOException;
+import java.util.Base64;
+
+public class SecretKeySerializer
+        extends JsonSerializer<SecretKey>
+{
+    @Override
+    public void serialize(SecretKey secretKey, JsonGenerator jsonGenerator, SerializerProvider serializerProvider)
+            throws IOException
+    {
+        jsonGenerator.writeString(Base64.getEncoder().encodeToString(secretKey.getEncoded()));
```

Comment:
technically `getEncoded` may return null. 

make it:
```java
        byte[] encodedBytes = secretKey.getEncoded();
        verify(encodedBytes != null, "Could not encode secret key");
        jsonGenerator.writeString(Base64.getEncoder().encodeToString(encodedBytes));
```

Reply:
Constructor of `SecretKeySpec` guarantees that it won't be null